Tasks/ServiceFabricDeploy/Tests/CreateDiffPkg.ps1

```diff
@@ -0,0 +1,101 @@
+Register-Mock Test-Path { $true } -- "HKLM:\SOFTWARE\Microsoft\Service Fabric SDK"
+
+# Setup mock VSTS service endpoint
+$vstsEndpoint = @{
+    "Auth" = @{
+        "Scheme" = "None"
+    }
+}
+Register-Mock Get-VstsEndpoint { $vstsEndpoint } -- -Name $serviceConnectionName -Require
+
+# Setup mock for connection to cluster
+Register-Mock Connect-ServiceFabricCluster { $null } -- -ConnectionEndpoint "test"
+
+# Setup mock registry settings
+$regKeyObj = @{
+    "FabricSDKPSModulePath" = $serviceFabricSdkModulePath
+}
+Register-Mock Get-ItemProperty { $regKeyObj } -- -Path "HKLM:\SOFTWARE\Microsoft\Service Fabric SDK" -Name FabricSDKPSModulePath
+
+Register-Mock Get-ApplicationNameFromApplicationParameterFile { $appName } -- "$PSScriptRoot\data\ApplicationParameters.xml"
+
+$app = @{
+    "ApplicationTypeName" = $applicationTypeName;
+    "ApplicationTypeVersion" = $applicationTypeVersion
+}
+Register-Mock Get-ServiceFabricApplication { $app } -- -ApplicationName $appName
+$publishArgs = @("-ApplicationParameterFilePath:", "$PSScriptRoot\data\ApplicationParameters.xml", "-OverwriteBehavior:", $overwriteBehavior, "-ApplicationPackagePath:", "aaa", "-ErrorAction:", "Stop", "-Action:", "RegisterAndCreate")
+Register-Mock Publish-NewServiceFabricApplication -Arguments $publishArgs
+
+$serviceType1 = @{
+    "ServiceManifestName" = "Stateless1Pkg"
+}
+$serviceType2 = @{
+    "ServiceManifestName" = "Stateless2Pkg"
+}
+$serviceTypes = @($serviceType1, $serviceType2)
+$serviceManifest = '<ServiceManifest Name="Stateless1Pkg" Version="1.0.0">' +
+  '<ServiceTypes>' +
+    '<StatelessServiceType ServiceTypeName="Stateless1Type" />' +
+  '</ServiceTypes>' +
+  '<CodePackage Name="Code" Version="1.0.0">' +
+  '</CodePackage>' +
+  '<ConfigPackage Name="Config" Version="1.0.0" />' +
+'</ServiceManifest>'
+
+Register-Mock Get-ServiceFabricServiceType {$serviceTypes} -- -ApplicationTypeName $applicationTypeName -ApplicationTypeVersion $applicationTypeVersion
+Register-Mock Get-ServiceFabricServiceManifest {$serviceManifest} -- -ApplicationTypeName $applicationTypeName -ApplicationTypeVersion $applicationTypeVersion -ServiceManifestName "Stateless1Pkg"
+Register-Mock Get-ServiceFabricServiceManifest {$serviceManifest} -- -ApplicationTypeName $applicationTypeName -ApplicationTypeVersion $applicationTypeVersion -ServiceManifestName "Stateless2Pkg"
+
+Register-Mock Copy-Item {} $appManifestPath $appManifestDiffPath -Force
+
+Microsoft.PowerShell.Core\Import-Module "$PSScriptRoot\..\Create-DiffPackage.psm1"
+
+# Act
+. $PSScriptRoot\..\..\..\Tasks\ServiceFabricDeploy\ps_modules\ServiceFabricHelpers\Connect-ServiceFabricClusterFromServiceEndpoint.ps1
+@( & $PSScriptRoot/../../../Tasks/ServiceFabricDeploy/deploy.ps1 )
+
+# Assert
+Assert-WasCalled Copy-Item $appManifestPath $appManifestDiffPath -Force
+Assert-WasCalled Copy-Item $serviceManifestPath2 $serviceManifestDiffPath2 -Force
+Assert-WasCalled Copy-Item $codePkg2 $codeDiffPkg2 -Recurse
```

Comment:
You need to also assert that service manifest 1 was not copied and that the deployment was run on the diff package not the full package,